lib/brexit_checker/criteria/filter.rb

```diff
@@ -0,0 +1,33 @@
+class BrexitChecker::Criteria::Filter
+  def call(selected_criteria)
+    loop do
+      old_size = selected_criteria.size
+      selected_criteria &= possible_criteria(selected_criteria)
+      raise "failsafe" unless selected_criteria.size <= old_size
```

Comment:
Can we have a explicit message the describes the error? - so we can easily identify the issue in logs etc.

Reply:
Sure. I was struggling with the text here, and was leaning towards keeping it short and relying on the code location feature we have in our logs (i.e. we'll know exactly where the exception came from, anyway). But I agree it would be difficult to search for. We could try: "failsafe: criteria should not increase in size". It's difficult to write about a future bug!

Reply:
```suggestion
      raise "Number of filtered criteria is larger than number of original criteria" unless selected_criteria.size <= old_size
```
Don't think the word "failsafe" adds any value.

Reply:
OK. I had another think and I've pushed a change that should now cover the 'what' and the 'why' more clearly - I reckon the use of the term 'assert' should help others understand the intention better.